kratos/geometries/coupling_geometry.h

```diff
@@ -337,6 +304,236 @@ template<class TPointType> class CouplingGeometry
         return mpGeometries[0]->Center();
     }
 
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    /// Provides the default integration per geometry.
+    IntegrationInfo GetDefaultIntegrationInfo() const override
+    {
+        const SizeType local_space_dimension = this->LocalSpaceDimension();
+
+        std::vector<SizeType> number_of_points_per_span_per_direction(local_space_dimension);
+        std::vector<IntegrationInfo::QuadratureMethod> quadrature_method_per_direction(local_space_dimension);
+        for (IndexType i = 0; i < local_space_dimension; ++i) {
+            SizeType max_p = 0;
+            for (IndexType j = 0; j < NumberOfGeometryParts(); ++j) {
+                max_p = std::max(mpGeometries[j]->PolynomialDegree(i) + 1, max_p);
+            }
+            number_of_points_per_span_per_direction[i] = max_p;
+            quadrature_method_per_direction[i] = IntegrationInfo::QuadratureMethod::GAUSS;
+        }
+
+        return IntegrationInfo(number_of_points_per_span_per_direction, quadrature_method_per_direction);
+    }
+
+    ///@}
+    ///@name Integration Points
+    ///@{
+
+    /* Creates integration points on the master considering all slave intersections.
+     * @return integration points.
+     */
+    void CreateIntegrationPoints(
+        IntegrationPointsArrayType& rIntegrationPoints,
+        IntegrationInfo& rIntegrationInfo) const override
+    {
+        const double model_tolerance = 1e-3;
+
+        if (this->Dimension() == 1) {
+            std::vector<double> intersection_master_spans;
+            std::vector<double> intersection_master_spans_on_master_spans;
+
+            mpGeometries[0]->SpansLocalSpace(intersection_master_spans);
+            CurveTessellation<PointerVector<TPointType>> curve_tessellation;
+            curve_tessellation.Tessellate(
+                *(mpGeometries[0].get()),
+                intersection_master_spans,
+                1e-2, mpGeometries[0]->PolynomialDegree(0), false);
+
+            CoordinatesArrayType local_coords_slave = ZeroVector(3);
+            CoordinatesArrayType global_coords = ZeroVector(3);
```

Comment:
Should be global_coords_slave, I guess.

Reply:
I renamed some variables. Hope it is more consistent now?